### ISSUE TITLE
Generic fota scraping script

### ISSUE THREAD

**Issue by haropng** [TBT8A10/adups-fota#3]
For many devices using adups, is there a universal script to achieve crawling OTA?

**Comment by TBM13** [comment]
I dont think so. To search for OTA updates you need a valid device info (model, locale, etc) and version.
Assuming you have the device info, you still need to get a valid version from the device itself or some dump available on internet, so there's no "universal script" to get all the OTA packages for a device

**Comment by haropng** [comment]
So is there a guide on how to decompile other devices to achieve the same effect as this repository?

**Comment by TBM13** [comment]
It's just literally grabbing these props from the device with `adb shell getprop`:
![image](https://github.com/user-attachments/assets/c582c6dd-0c4c-4a75-851e-17356ddc430d)

**Comment by haropng** [comment]
![IMG_20240923_231721_354](https://github.com/user-attachments/assets/9186774e-0384-4a78-bef5-f88b7041ee8a)
I fetched the update but when I open the link in my browser I get this screen

**Comment by TBM13** [comment]
Weird. Can you send me the props so I can try it myself?

**Comment by haropng** [comment]
When I use 1dm to download, it prompts HTTP/1.1 403 Forbidden
https://github.com/harojpgyy/Get_adups_fota/blob/master/ota.py

**Comment by TBM13** [comment]
I don't know what 1dm is. I just tried to download it and I get the same issue.
Looks like an issue on their server, there's nothing I can do

**Comment by haropng** [comment]
https://github.com/harojpgyy/Get_FOTA/tree/abupdate.fota_demo_iot
The new fota, even though it is very similar to adups, it is not from the same vendor, the adups script is no longer applicable, if you are interested, I uploaded the ota program and official demo of a certain device

**Comment by TBM13** [comment]
I took a look to `HescOta.apk` and at a first glance it seems to be using "abupdate" and not ADUPS (although it does seem to support adups too).
![image](https://github.com/user-attachments/assets/663a1953-0562-4e1c-a302-c2a4914eb272)

Either way, it seems to have some differences so I doubt my Python script can easily be adapted to support it. Hard to tell without having a request made by the app at hand.

**Comment by haropng** [comment]
It seems that it no longer uses those properties in build.prop, but takes a specific "mid", in this case, the mid is imei

**Comment by haropng** [comment]
https://fota5p.adups.com/otainter-5.0/fota5/detectSchedule.do 
How is this server address obtained?

**Comment by ikheeteric** [comment]
Any solutions?

**Comment by ikheeteric** [comment]
const RO_FOTA_OEM = 'lovdream8953_8.0';
const RO_FOTA_DEVICE = 'UT30';
const RO_PRODUCT_LOCALE = 'en-US';
const RO_OPERATOR_OPTR = '';
const RO_FOTA_VERSION = 'R00.52.40.40_20180621-1124'; i have this but don't get any firmware

**Comment by haropng** [comment]
> const RO_FOTA_OEM = 'lovdream8953_8.0'; const RO_FOTA_DEVICE = 'UT30' ; const RO_PRODUCT_LOCALE = 'en-US'; const RO_OPERATOR_OPTR = ''; const RO_FOTA_VERSION = 'R00.52.40.40_20180621-1124'; 我有这个，但没有得到任何固件

It looks like the firmware is from June 21, 2018. As the readme says, your version is too old and the device may have been deleted from the server.

**Comment by amoscatelli** [comment]
Hi @TBM13 @haropng,
thank you for these repositories.
I am trying to fetch the firmware for my device (I want to try and extract the boot.img from that) : 

RO_FOTA_OEM = 'chiptrip_AW_11.0'
RO_FOTA_DEVICE = 'F800'
RO_PRODUCT_LOCALE = 'en-US'
RO_OPERATOR_OPTR = ''

RO_FOTA_VERSION = 'F800_Android11_202503010959_20250301-0959'

Problem is, I already upgraded from the device so I don't know/remember the previous RO_FOTA_VERSION.

Any suggestion ?

![Image](https://github.com/user-attachments/assets/eeaa04d9-69e1-42f4-8fa9-0e3760f65090)

I hope I don't need to wait for the next update

**Comment by TBM13** [comment]
> Problem is, I already upgraded from the device so I don't know/remember the previous RO_FOTA_VERSION.

Only thing I can think of is to bruteforce the value of ro.fota.version, or grabbing it from a device that hasn't updated yet.

**Comment by haropng** [comment]
lts boot may be just incremental, you can use other tools to extract the complete firmware

**Comment by amoscatelli** [comment]
I found out the previous firmware version : F800_Android11_202412031735_20241203-1735

The script worked : 

```
C:\Users\drago\Get_FOTA>python3 ota.py
[*] Config: {'project': 'chiptrip$AW$11.0_F800_en-US_other', 'version': 'F800_Android11_202412031735_20241203-1735'}
[*] POST Data: key=9C080000000000000000BF6288024189594226ABB57DD13C632F4254483CF916DEC68738632958156A1FE0448781C71357781801742DAA58E3FCA82365284D430D3EA107C5C698222C0610011B17851BD2DD98257571196E1978F641879DC77F20771B04747AF447829EC57C226D1906187D&shaKey=3028e190542463785d98f3b3ac6abd38b8bc67f047f1584fee2c21c514dd4103
[*] POST: https://fota5p.adups.com/otainter-5.0/fota5/detectSchedule.do
[+] Update found!
[Update Info]
   Version: F800_Android11_202503010959_20250301-0959
   Size: 78.815609 MB
   Download URL: http://cloudflare.mayitek.com/ota/root_data02_2/chiptrip/chiptrip_AW_11.0/F800/en-US/other/F800_Android11_202503010959_20250301-0959/2025030110340386829.zip?m=null
   MD5 Sum: e32f2ca1fcc4733901b00c7ab1f521c2
```

I managed to download the zip but it seems it contains only the differential OTA so I don't think I can extract the boot.img :(

**Comment by TBM13** [comment]
> I found out the previous firmware version : F800_Android11_202412031735_20241203-1735
> 
> The script worked :
> 
> ```
> C:\Users\drago\Get_FOTA>python3 ota.py
> [*] Config: {'project': 'chiptrip$AW$11.0_F800_en-US_other', 'version': 'F800_Android11_202412031735_20241203-1735'}
> [*] POST Data: key=9C080000000000000000BF6288024189594226ABB57DD13C632F4254483CF916DEC68738632958156A1FE0448781C71357781801742DAA58E3FCA82365284D430D3EA107C5C698222C0610011B17851BD2DD98257571196E1978F641879DC77F20771B04747AF447829EC57C226D1906187D&shaKey=3028e190542463785d98f3b3ac6abd38b8bc67f047f1584fee2c21c514dd4103
> [*] POST: https://fota5p.adups.com/otainter-5.0/fota5/detectSchedule.do
> [+] Update found!
> [Update Info]
>    Version: F800_Android11_202503010959_20250301-0959
>    Size: 78.815609 MB
>    Download URL: http://cloudflare.mayitek.com/ota/root_data02_2/chiptrip/chiptrip_AW_11.0/F800/en-US/other/F800_Android11_202503010959_20250301-0959/2025030110340386829.zip?m=null
>    MD5 Sum: e32f2ca1fcc4733901b00c7ab1f521c2
> ```
> 
> I managed to download the zip but it seems it contains only the differential OTA so I don't think I can extract the boot.img :(

Ah, right. Iirc the only complete image the OTAs include is uboot. Does your device have a locked bootloader? If not, you could just patch uboot and dump boot in loader mode

**Comment by amoscatelli** [comment]
Developer Options say the bootloader is unlocked, can this be trusted ?
Do you know a guide or tutorial to batch and boot uboot ?

**Comment by TBM13** [comment]
> Developer Options say the bootloader is unlocked, can this be trusted ? Do you know a guide or tutorial to batch and boot uboot ?

No, that doesn't really mean anything (at least on my device). I suggest you to [read this](https://github.com/TBT8A10/Research).
